### DISCUSSION TITLE
Create testing instances before deploying to main

### DISCUSSION THREAD

**Discussion by gustavovalverde** [ZcashFoundation/zebra#3670] [Ideas]
## Motivation

- Some issues might arise just after a successful deployment to the infrastructure (GCP so far)
- Local testing environment and the production environment might differ
- We might want to evaluate the "actual" time a PR might take to sync on the blockchain up to the tip or any similar behavior

### Designs

- When a PR passes the CI, deploy a testing instance and report back to GitHub with a health-check to confirm it was successfully deployed and everything is working as expected.
- If this Integration Test fails, we must validate before reviewing/merging to the main branch

**Comment by conradoplg** [comment]
> * If this Integration Test fails, we must validate before reviewing/merging to the main branch

One issue is that sync takes hours, so it could make things difficult when merging a bunch of PRs one after another.
Not sure if we should just accept it, or maybe allowing merging to main (as long as the other tests pass) but running the sync test and notifying if it breaks, so that we can prioritize fixing it.

**Comment by conradoplg** [comment]
@gustavovalverde @teor2345  Just to check, this is blocked on us trying to make sync faster and deciding to enforce this on PRs, or to just to test it asynchronously, right?

**Comment by teor2345** [comment]
I think this extra testing is a good idea, but there are other things that are a higher priority, like:
- #3510 
- #3391

**Comment by mpguerra** [comment]
@gustavovalverde did we want to do anything about this? Can we close this discussion?
